src/liballoc/boxed.rs

```diff
@@ -276,6 +276,13 @@ impl<T: fmt::Debug + ?Sized> fmt::Debug for Box<T> {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
+impl<T> fmt::Pointer for Box<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Pointer::fmt(&*self, f)
```

Comment:
Oops. Is `Box` specialcased in some way in the compiler? I can't seem to extract the `Unique` within using tuple access or a destructuring match.

```
expected `&Box<T>`,
    found `boxed::Box<_>`
(expected &-ptr,
    found struct `boxed::Box`) [E0308]
/Users/richo/code/ext/rust/src/liballoc/boxed.rs:282             Box(uniq) => fmt::Pointer::fmt(&uniq, f)
```